subprojects/core/src/main/java/org/gradle/internal/service/scopes/ProjectScopeServices.java

```diff
@@ -373,4 +374,9 @@ protected ManagedFactoryRegistry createManagedFactoryRegistry(ManagedFactoryRegi
             new org.gradle.api.internal.file.ManagedFactories.DirectoryPropertyManagedFactory(filePropertyFactory)
         );
     }
+
+    @Provides
+    protected InitProjectSpecLoader createInitProjectSpecLoader(ProjectInternal project) {
+        return new InitProjectSpecLoader(project.getClassLoaderScope().getLocalClassLoader(), project.getLogger());
```

Comment:
So the problem is we can't store a project or a classLoaderScope reference in the task itself, since these types aren't serializable with the CC.  We need to access the local class loader at or after task construction time, to load any specs.  

Defering accessing the project's class loader scope until task execution time works and removes the need for `withEagerClassLoaderCreationCheckDisabled` in `BuildScanScopeIdsIntegrationTest`...but we shouldn't access the project during execution.

We can't just move the loading into a project-scoped service.  I've been experimenting a lot with this, and can't find any way past this.  It would be great if we could load the registry in a service and have the task only _access_ it...but this doesn't seem to be possible.  Services seems to be created too soon - the plugins won't be on the project's classpath yet, so they won't be loaded.

I'm not sure how to move forward on changing this to avoid class loader problems Paul mentions in https://github.com/gradle/gradle/pull/30098#discussion_r1716832927.  I'm also not sure that we are actually going to prevent a switch to strict classloading with this change.  So, really just looking for ideas at this point.